src/e2e/puppeteer/__tests__/caret.ts

```diff
@@ -214,8 +214,13 @@ describe('all platforms', () => {
     await press('Escape')
     await press('ArrowDown')
 
-    const textContext = await getSelection().focusNode?.textContent
-    expect(textContext).toBe('a')
+    // Wait for the caret to move to the editable
+    // because the closing of command palette is asynchronous and the caret may not be in the editable yet
+    // otherwise the test intermittently fails in CI.
+    await waitUntil(() => window.getSelection()?.focusNode?.textContent === 'a')
```

Comment:
Let's abstract this out into a helper since it's being used in multiple places. It'll make the tests a bit more readable, and allow us to move the extensive commentary to the helper.

`waitForSelectionContent`